code/parse/sexp.cpp

```diff
@@ -23339,95 +23169,88 @@ int sexp_script_eval(int node, int return_type, bool concat_args = false)
 
 void sexp_script_eval_multi(int node)
 {
-	char s[TOKEN_LENGTH];
+	char script[TOKEN_LENGTH];
 	bool success = true;
 	bool execute_on_server;
-	int sindex;
 	player *p;
 
-	strcpy_s(s, CTEXT(node));
-
+	strcpy_s(script, CTEXT(node));
 	node = CDR(node);
 
 	execute_on_server = is_sexp_true(node);
-
 	node = CDR(node);
 
 	Current_sexp_network_packet.start_callback();
-	Current_sexp_network_packet.send_string(s);
+	Current_sexp_network_packet.send_string(script);
 	// evalutate on all clients
 	if (node == -1) {
 		Current_sexp_network_packet.send_bool(true);			
-		execute_on_server = 1;
+		execute_on_server = true;
 	}
 	// we have to send to all clients but we need to send a list of ships so that they know if they evaluate or not
 	else {
 		Current_sexp_network_packet.send_bool(false);
 
-		do {
+		for (; node != -1; node = CDR(node)) {
 			p = get_player_from_ship_node(node, true);
 
 			// not a player ship so skip it
 			if (p == NULL ){
-				node = CDR(node);
 				continue;
 			}
 			else {
 				// if this is me, flag that we should execute the script
 				if (p == Player) {					
-					execute_on_server = 1;
+					execute_on_server = true;
 				}
 				// otherwise notify the clients
 				else {
-					sindex = ship_name_lookup(CTEXT(node));
-					Current_sexp_network_packet.send_ship(sindex);
+					Current_sexp_network_packet.send_string(CTEXT(node));
 				}
 			}
-
-			node = CDR(node);
-		} while (node != -1); 
+		}
 	}
 
 	Current_sexp_network_packet.end_callback();
 
 	if (execute_on_server) {		
-		success = Script_system.EvalString(s, s);
+		success = Script_system.EvalString(script, script);
 	}
 
 	if(!success) {
-		Warning(LOCATION, "sexp-script-eval failed to evaluate string \"%s\"; check your syntax", s);
+		Warning(LOCATION, "sexp-script-eval failed to evaluate string \"%s\"; check your syntax", script);
 	}
 }
 
 void multi_sexp_script_eval_multi()
 {
-	int sindex;
-	char s[TOKEN_LENGTH];
+	char script[TOKEN_LENGTH];
+	char ship_name[TOKEN_LENGTH];
 	bool sent_to_all = false;
 	bool success = true;
 
-	Current_sexp_network_packet.get_string(s);
+	Current_sexp_network_packet.get_string(script);
 	Current_sexp_network_packet.get_bool(sent_to_all);
 
 	if (sent_to_all) {
-		success = Script_system.EvalString(s, s);
+		success = Script_system.EvalString(script, script);
 	}
 	// go through all the ships that were sent and see if any of them match this client.
 	else {
-		while (Current_sexp_network_packet.get_ship(sindex)) {
-			Assertion(sindex >= 0, "Illegal value for the ship index sent in multi_sexp_script_eval_multi()! Ship %d does not exist!", sindex); 
-			if (Player->objnum == Ships[sindex].objnum) {
-				success = Script_system.EvalString(s, s);
+		while (Current_sexp_network_packet.get_string(ship_name)) {
+			auto ship_entry = ship_registry_get(ship_name);
+			Assertion(ship_entry, "Illegal value for the ship index sent in multi_sexp_script_eval_multi()! Ship %s does not exist!", ship_name);
```

Comment:
Minor thing: It says `ship index` here but we are looking at a ship name so that could be adjusted here for consistency.

Reply:
Good point.